libfwupdplugin/fu-sbatlevel-section.c

```diff
@@ -72,40 +72,69 @@ fu_sbatlevel_section_parse(FuFirmware *firmware,
 			   FwupdInstallFlags flags,
 			   GError **error)
 {
-	gsize header_offset = offset + FU_STRUCT_SBAT_LEVEL_SECTION_HEADER_OFFSET_PREVIOUS;
-	guint32 previous_addr;
-	guint32 latest_addr;
 	g_autoptr(GByteArray) st = NULL;
 
 	st = fu_struct_sbat_level_section_header_parse_stream(stream, offset, error);
 	if (st == NULL)
 		return FALSE;
-
-	previous_addr = fu_struct_sbat_level_section_header_get_previous(st);
-
-	if (!fu_sbatlevel_section_add_entry(firmware,
-					    stream,
-					    header_offset + previous_addr,
-					    "previous",
-					    0,
-					    flags,
-					    error))
+	if (!fu_sbatlevel_section_add_entry(
+		firmware,
+		stream,
+		offset + fu_struct_sbat_level_section_header_get_previous(st),
```

Comment:
I am a bit confused here. I thought that the "previous" offset was relative to after the header (that is the 2 offsets in the beginning). But here w add it to `offset`, which I would think is the beginning of the section. Or am I missing something?

Reply:
No, that's exactly why I asked you! I thought the offset was from the start. Do you have a known-good sbatlevel blob handy you could upload so I can try to recreate it please? Thanks!

Reply:
Also, confusingly, the existing code uses FU_STRUCT_SBAT_LEVEL_SECTION_HEADER_OFFSET_PREVIOUS which is 0x4, i.e. neither the start or the end of the struct!

Reply:
Actually it is relative the offset table. Which comes after the version. So `FU_STRUCT_SBAT_LEVEL_SECTION_HEADER_OFFSET_PREVIOUS` was correct. It is only the version we have to skip.

See https://github.com/rhboot/shim/blob/5914984a1ffeab841f482c791426d7ca9935a5e6/sbat_var.S
The addresses  are  `sbat_var_automatic - sbat_var_payload_header` and `sbat_var_latest - sbat_var_payload_header`. And `sbat_var_payload_header` is just after the version.


Here is an example:
```
./usr/lib/shim/shimx64.efi.dualsigned:     file format pei-x86-64

Contents of section .sbatlevel:
 87000 00000000 08000000 22000000 73626174  ........"...sbat
 87010 2c312c32 30323230 35323430 300a6772  ,1,2022052400.gr
 87020 75622c32 0a007362 61742c31 2c323032  ub,2..sbat,1,202
 87030 32313131 3530300a 7368696d 2c320a67  2111500.shim,2.g
 87040 7275622c 330a00                      rub,3..         
```

Reply:
github does not let me upload a small binary file. You can base64 decode this if you want:

`AAAAAAgAAAAiAAAAc2JhdCwxLDIwMjIwNTI0MDAKZ3J1YiwyCgBzYmF0LDEsMjAyMjExMTUwMApzaGltLDIKZ3J1YiwzCgA=`

Reply:
Super useful, thanks! PR updated -- does that look okay now?

Reply:
It does.